### ISSUE TITLE
support for separate server/client specific vite configurations?

### ISSUE THREAD

**Issue by daniellacosse** [nuxt/nuxt#19610]
### Describe the feature

Thanks to the Nuxt3 team for making such an incredible framework! Admittedly I have a very narrow use case - I'm using the amazingly awesome [rollup-obfuscator](https://www.npmjs.com/package/rollup-obfuscator) and to my surprise it actually mostly works - but certain features break the compilation process. 

For example, if I use [`selfDefending`](https://github.com/javascript-obfuscator/javascript-obfuscator#selfdefending), during generation the prerender step hangs forever because I assume we're using `vite` to prepare some code that then gets _further_ modified in _that_ step - this further modification triggering the self-defense applied, stalling out the process. Super cool to see the obfuscation working as intended, but I really only want to use the obfuscator in the final compilation step anyway.

TL;DR - Having a place to specify a client-specific vite config would I think solve this? And may have other applications.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module? _(I'm not sure 😅)_

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment]
You can do this at the moment with a Nuxt hook:

```js
export default defineNuxtConfig({
  hooks: {
    'vite:extendConfig': (config, { isClient, isServer }) => {
      // Extend vite config here
    }
  }
})
```

**Comment by Ahidada** [comment]
Here's an example.

`yarn add javascript-obfuscator -D`
`yarn add rollup-plugin-obfuscator -D`

> file: nuxt.config.js
`import obfuscatorBox from 'rollup-plugin-obfuscator';

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  // ssr: false,
  devtools: { enabled: true },
  modules: [
    '@nuxt/content'
  ],
  nitro: {
    prerender: {
      // routes: ['/hello']
      routes: ['/']
      // ignore: ['/dynamic']
    }
  },
  content: {
    // https://content.nuxtjs.org/api/configuration
    // api: {
    //   baseURL: '/api/_my_content'
    // }
    documentDriven: true,
    highlight: {
      theme: 'one-dark-pro'
    }
  },
  hooks: {
    'vite:extendConfig': (config, { isClient, isServer }) => {
      // Extend vite config here
      // console.log("config.plugins =", config.plugins);
      if(isClient) {
        config.plugins.push(
          obfuscatorBox({
            global:true,
            options: {
              compact: true,
              controlFlowFlattening: true,
              controlFlowFlatteningThreshold: 0.75,
              numbersToExpressions: true,
              simplify: true,
              stringArrayShuffle: true,
              splitStrings: true,
              splitStringsChunkLength: 10,
              rotateUnicodeArray: true,
              deadCodeInjection: true,
              deadCodeInjectionThreshold: 0.4,
              debugProtection: false,
              debugProtectionInterval: 2000,
              disableConsoleOutput: true, 
              domainLock: [],
              identifierNamesGenerator: "hexadecimal",
              identifiersPrefix: "",
              inputFileName: "",
              log: true,
              renameGlobals: true,
              reservedNames: [],
              reservedStrings: [],
              seed: 0,
              selfDefending: true,
              sourceMap: false,
              sourceMapBaseUrl: "",
              sourceMapFileName: "",
              sourceMapMode: "separate",
              stringArray: true,
              stringArrayEncoding: ["base64"],
              stringArrayThreshold: 0.75,
              target: "browser",
              transformObjectKeys: true,
              unicodeEscapeSequence: true,
  
  
              domainLockRedirectUrl: "about:blank",
              forceTransformStrings: [],
              identifierNamesCache: null,
              identifiersDictionary: [],
              ignoreImports: true,
              optionsPreset: "default",
              renameProperties: false,
              renamePropertiesMode: "safe",
              sourceMapSourcesMode: "sources-content",
  
              stringArrayCallsTransform: true,
              stringArrayCallsTransformThreshold: 0.5,
  
              stringArrayIndexesType: ["hexadecimal-number"],
              stringArrayIndexShift: true,
              stringArrayRotate: true,
              stringArrayWrappersCount: 1,
              stringArrayWrappersChainedCalls: true,
              stringArrayWrappersParametersMaxCount: 2,
              stringArrayWrappersType: "variable",
            },
          }),
        )
      }
    }
  }
})
`

**Comment by danielroe** [comment]
This is actually implemented with https://github.com/nuxt/nuxt/pull/22304 and https://github.com/nuxt/nuxt/pull/22302.
